src/Game/UI/Controls/StbTextBoxCombo.cs

```diff
@@ -0,0 +1,142 @@
+﻿#region license
+// Copyright (C) 2020 ClassicUO Development Community on Github
+// 
+// This project is an alternative client for the game Ultima Online.
+// The goal of this is to develop a lightweight client considering
+// new technologies.
+// 
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+// 
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#endregion
+
+using ClassicUO.Game.Managers;
+using ClassicUO.Input;
+using ClassicUO.IO.Resources;
+using ClassicUO.Renderer;
+using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ClassicUO.Game.UI.Controls
+{
+    internal class StbTextBoxCombo : StbTextBox
+    {
+        private string[] _items;
+        private int _selectedIndex;
+        private GumpPic _arrow;
+        private readonly byte _font;
+        private readonly int _maxWidth;
+        private readonly int _maxComboHeight;
+        private ComboboxContextMenu _contextMenu;
+        private const ushort ARROW_UP = 253;
+        private const ushort ARROW_DOWN = 252;
+
+        public StbTextBoxCombo(string[] items, byte font, int max_char_count = -1, int maxWidth = 0, bool isunicode = true, FontStyle style = FontStyle.None, ushort hue = 0, TEXT_ALIGN_TYPE align = TEXT_ALIGN_TYPE.TS_LEFT, int maxComboHeight = 100) :
+            base(font, max_char_count, maxWidth, isunicode, style, hue, align)
+        {
+            _maxWidth = maxWidth;
+            _maxComboHeight = maxComboHeight;
+            _font = font;
+            SetItems(items);
+        }
+
+        internal void SetItems(string[] items)
+        {
+            _items = items;
+            if (_items != null && _items.Length > 0 && _arrow == null)
+            {
+                Add(_arrow = new GumpPic(_maxWidth - 6, 4, ARROW_UP, 0));
+            }
+        }
+
+        public int SelectedIndex
+        {
+            get => _selectedIndex;
+            set
+            {
+                _selectedIndex = value;
```

Comment:
Maybe
```cs
if (_selectedIndex == value || value < 0)
{
    return;
}
...
```

Reply:
indeed.

Reply:
Actually. This was also some copy pasta from the `ComboBox` and now that i look at it...  None of this logic should really reside in this setter.  This is probably better handled in the event handler for the `OnItemSelected` event that i added to the `ComboboxContextMenu`.

Reply:
I have removed the auto property and moved all the relevant functionality to the event handler.